src/access/acp.test.ts

```diff
@@ -4612,6 +4613,273 @@ describe("setActorAccess", () => {
+        fc.constant("https://some.pod/resource?ext=acl#rule1"),
+        fc.constant("https://some.pod/resource?ext=acl#rule2"),
+        fc.constant("https://some.pod/resource?ext=acl#rule3")
+      );
+      const agentUrlArbitrary = fc.oneof(
+        fc.constant("https://some.pod/profiles#agent1"),
+        fc.constant("https://some.pod/profiles#agent2"),
+        fc.constant("https://some.pod/profiles#agent3"),
+        fc.constant(acp.PublicAgent),
+        fc.constant(acp.AuthenticatedAgent),
+        fc.constant(acp.CreatorAgent)
+      );
+      const groupUrlArbitrary = fc.oneof(
+        fc.constant("https://some.pod/groups#group1"),
+        fc.constant("https://some.pod/groups#group2"),
+        fc.constant("https://some.pod/groups#group3")
+      );
+      const ruleArbitrary = fc.record({
+        [acp.agent]: fc.option(fc.array(agentUrlArbitrary, { maxLength: 6 }), {
+          nil: undefined,
+        }),
+        [acp.group]: fc.option(fc.array(groupUrlArbitrary, { maxLength: 3 }), {
+          nil: undefined,
+        }),
+      });
+      const policyAccessArbitrary = fc.dictionary(
+        fc.oneof(fc.constant("allow"), fc.constant("deny")),
+        mockAccessArbitrary
+      );
+      const policyRulesArbitrary = fc.dictionary(
+        fc.oneof(
+          fc.constant("allOf"),
+          fc.constant("anyOf"),
+          fc.constant("noneOf")
+        ),
+        fc.dictionary(ruleUrlArbitrary, ruleArbitrary)
+      );
+      const policyArbitrary = fc
+        .tuple(policyAccessArbitrary, policyRulesArbitrary)
+        .map(([access, rules]) => ({ ...access, ...rules }));
+      const acrArbitrary = fc.record({
+        policies: fc.oneof(
+          fc.constant({}),
+          fc.dictionary(policyUrlArbitrary, policyArbitrary)
+        ),
+        memberPolicies: fc.constant({}),
+        acrPolicies: fc.oneof(
+          fc.constant({}),
+          fc.dictionary(policyUrlArbitrary, policyArbitrary)
+        ),
+        memberAcrPolicies: fc.constant({}),
+      });
+      const actorRelationArbitrary = fc.oneof(
+        fc.constant(acp.agent),
+        fc.constant(acp.group)
+      );
+      const fcInput = fc.tuple(
+        acrArbitrary,
+        setAccessArbitrary,
+        actorRelationArbitrary,
+        agentUrlArbitrary
+      );
+      const fcResult = fc.check(
+        fc.property(
+          fcInput,
+          ([acrConfig, accessToSet, actorRelation, actorUrl]) => {
+            const resourceWithAcr = mockResourceWithAcr(
+              "https://some.pod/resource",
+              "https://some.pod/resource?ext=acr",
+              acrConfig as any
+            );
+            const updatedResource = internal_setActorAccess(
+              resourceWithAcr,
+              actorRelation,
+              actorUrl,
+              accessToSet
+            );
+            expect(
+              internal_getActorAccess(updatedResource!, actorRelation, actorUrl)
+            ).toStrictEqual(accessToSet);
```

Comment:
This is the actual interesting part; So given a randomly generated pre-existing ACR configuration, and a randomly generated object of access to set, it verifies that the return value of `getActorAccess` actually was what was passed to `setActorAccess`.

As can be seen in the other skipped unit tests below, there are some ACR configurations in which that is not the case, unfortunately.